lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
@afchung , yes this is what I also checked. I am not able to access it.

On Tue, Sep 29, 2015 at 10:36 AM, Andrew Chung <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40702238>:
>
> >              {
> > +                GC.Collect();
>
> To clarify, this
> <https://msdn.microsoft.com/en-us/library/system.runtime.gcsettings.largeobjectheapcompactionmode(v=vs.110).aspx>
> setting is the one I'm referring to. The GCSettings
> <https://msdn.microsoft.com/en-us/library/system.runtime.gcsettings(v=vs.110).aspx>
> documentation seems to say that it's present in .NET 4.6, would be strange
> if it's not there...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40702238>.
>